src/dht/dht_tracker.cc

```diff
@@ -47,7 +47,7 @@ DhtTracker::get_peers(unsigned int maxPeers) {
     throw internal_error("DhtTracker::BencodeAddress is packed incorrectly.");
 
   auto first = m_peers.begin();
-  auto last  = m_peers.end();
+  auto last  = std::prev(m_peers.end());
```

Comment:
This looks wrong, we should rather fix how the second argument to raw_list is derived when `last == end()`.

Reply:
I moved this change to the end. Not sure if this is what you meant.

Reply:
No, you need to change how the `bencode()` distance is calculated, without changing `last`.

It should be `std::distance(first, last) * sizeof(BencodeAddress)` afaics.